### ISSUE TITLE
Show a meaningful error message when server returns a non-json response (or ignore until happens more than once?)

### ISSUE THREAD

**Issue by tasn** [etesync/android#42]
Example:

```
EXCEPTION:
com.google.gson.JsonParseException: Expecting object found: \"<html>\"
	at com.google.gson.JsonObjectDeserializationVisitor.visitFieldUsingCustomHandler(JsonObjectDeserializationVisitor.java:100)
	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
	at com.google.gson.JsonDeserializationContextDefault.fromJsonPrimitive(JsonDeserializationContextDefault.java:85)
	at com.google.gson.JsonDeserializationContextDefault.deserialize(JsonDeserializationContextDefault.java:56)
	at com.google.gson.Gson.fromJson(Gson.java:551)
	at com.google.gson.Gson.fromJson(Gson.java:498)
	at com.google.gson.Gson.fromJson(Gson.java:441)
	at com.etesync.syncadapter.journalmanager.BaseManager.newCall(BaseManager.java:35)
	at com.etesync.syncadapter.journalmanager.JournalManager.list(JournalManager.java:47)
	at com.etesync.syncadapter.syncadapter.SyncAdapterService$SyncAdapter$RefreshCollections.run(SyncAdapterService.java:153)
	at com.etesync.syncadapter.syncadapter.CalendarsSyncAdapterService$SyncAdapter.onPerformSync(CalendarsSyncAdapterService.java:72)
	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:272)

```

Should handle the exception gracefully and return a meaningful error.

**Comment by tasn** [comment]
Related to #39

**Comment by tasn** [comment]
Fixed in EteSync 2.0
